cpp/src/strings/convert/convert_durations.cu

```diff
@@ -84,10 +84,10 @@ struct alignas(4) format_item {
  * components and when formatting a string from duration components.
  */
 struct format_compiler {
-  std::string format;
+  std::string_view const format;
   rmm::device_uvector<format_item> d_items;
-  format_compiler(const char* format_, rmm::cuda_stream_view stream)
-    : format(format_), d_items(0, stream)
+  format_compiler(std::string_view format, rmm::cuda_stream_view stream)
+    : format(format), d_items(0, stream)
```

Comment:
Should this be renamed to avoid the `format(format)` assignment?

Reply:
No, the compiler handles this fine and is supported by the C++ standard (section 12.6.2 I believe).
Using the same name where the argument directly sets a member variable in the ctor initializer reinforces they are literally the same thing.

Reply:
I know it is technically valid, but I thought the guide was to use the `format_` nomenclature. I'm not strongly opinionated here though if you'd like to keep it.

Reply:
I absolutely would not call it necessary.